x/evm/keeper/grpc_query.go

```diff
@@ -393,3 +400,100 @@ func (k Keeper) EthCall(c context.Context, req *types.EthCallRequest) (*types.Ms
 
 	return res, nil
 }
+
+// EstimateGas implements eth_estimateGas rpc api.
+func (k Keeper) EstimateGas(c context.Context, req *types.EthCallRequest) (*types.EstimateGasResponse, error) {
```

Comment:
can you add tests for this on a follow-up PR?

Reply:
I tried to do it, but there's so many context prepare things to do in unit test environment, I'm not sure how to do it.
1. With the default `KeeperTestSuite`, `GetCoinbaseAddress` will raise an error
2. We need to allocate some funds in genesis.